Lib/test/test_dataclasses.py

```diff
@@ -3808,6 +3828,18 @@ class A:
         with self.assertRaisesRegex(TypeError, msg):
             A(3, 4, 5)
 
+    def test_KW_ONLY_annotated(self):
+        @dataclass
+        class A:
+            a: int
+            _: Annotated[KW_ONLY, 'meta']
```

Comment:
I'm not sure this should be supported.

Reply:
I'm not sure either; it's a bit weird. Let me try to argue for it and if I can't convince you I'll remove it:

This is only a pseudo-field so it's a bit weird to annotate, but I think it's thematically okay to just allow Annotated to be put everywhere to avoid being completely agnostic to what someone wants to use it Annotated for. The most conservative move we can make is to make sure it's working for all field annotations, even pseudo ones like this. On the more weird side of things, it could be someone might want to try to use this symbol to extend dataclasses to do something else.

Are you convinced, or is it still too strange and I should remove it?

In favor of removing it, annotating `*` in a function is invalid syntax.

Reply:
If someone comes up with a concrete use case, I might reconsider, but it's documented as being a special sentinel value, not a real type annotation. I'd rather keep it simple than complicate it.

Reply:
All right, I'll remove it. This simplifies the other code as well since we don't need to strip Annotated before the KW_ONLY part of the code.